### ISSUE TITLE
Using PixelCopy in conjunction with ShadowPackageManager.addActivityIfNotPresent and robolectric.screenshot.hwrdr.native results in capturing an empty image.

### ISSUE THREAD

**Issue by takahirom** [robolectric/robolectric#8982]
### Description
While using a Manifest file to add an Activity allows for successful screenshot captures, employing ShadowPackageManager.addActivityIfNotPresent results in a transparent image instead of the expected bitmap.

### Steps to Reproduce

Use following settings and code.

* robolectric.screenshot.hwrdr.native=true
* ShadowPackageManager.addActivityIfNotPresent
* PixelCopy.request

Please add this test into ActivityScenarioTest in Robolectric repo to reproduce the issue

```kotlin
  static class CustomActivity extends Activity {
    @Override public void onCreate(Bundle savedInstanceState) {
      super.onCreate(savedInstanceState);
      View view = new View(this);
      view.setBackgroundColor(0xffff0000);
      view.layout(0, 0, 100, 100);
      setContentView(view);
    }
  }

  @GraphicsMode(GraphicsMode.Mode.NATIVE) @Test
  public void hasPixelWhenUsingCustomAddingActivityHardwareRendering() {
    // **If we use false here, the test will pass.**
    System.setProperty("robolectric.screenshot.hwrdr.native", "true");
    Application application = ApplicationProvider.getApplicationContext();
    ((ShadowPackageManager) Shadows.shadowOf(
        application.getPackageManager())).addActivityIfNotPresent(
        new Intent(application, CustomActivity.class).resolveActivity(
            application.getPackageManager()));

    try (ActivityScenario<CustomActivity> activityScenario = ActivityScenario.launch(
        CustomActivity.class)) {
      activityScenario.onActivity(activity -> {
        Window window = activity.getWindow();
        View decorView = window.getDecorView();
        Espresso.onIdle();
        Bitmap viewDrawBitmap = Bitmap.createBitmap(decorView.getWidth(), decorView.getHeight(),
            Bitmap.Config.ARGB_8888);
        decorView.draw(new Canvas(viewDrawBitmap));

        Bitmap pixelCopyBitmap = Bitmap.createBitmap(decorView.getWidth(), decorView.getHeight(),
            Bitmap.Config.ARGB_8888);
        CountDownLatch latch = new CountDownLatch(1);
        PixelCopy.request(window, pixelCopyBitmap, copyResult -> latch.countDown(),
            new Handler(Looper.getMainLooper()));
        try {
          latch.await(1, TimeUnit.SECONDS);
        } catch (InterruptedException e) {
          throw new RuntimeException(e);
        }
        int pixelCopyBitmapPixel = pixelCopyBitmap.getPixel(0, 0);

        int viewDrawBitmapPixel = viewDrawBitmap.getPixel(0, 0);
        System.out.println("viewDrawBitmapPixel: " + Integer.toHexString(viewDrawBitmapPixel));
        System.out.println("pixelCopyBitmapPixel: " + Integer.toHexString(pixelCopyBitmapPixel));
        assertThat(pixelCopyBitmapPixel).isEqualTo(viewDrawBitmapPixel);
        pixelCopyBitmap.recycle();
      });
    }
  }
```

Result

```
> Task :integration_tests:androidx_test:testDebugUnitTest
viewDrawBitmapPixel: ff191c1e
pixelCopyBitmapPixel: 0

expected: -15131618
but was : 0
expected: -15131618
but was : 0
	at org.robolectric.integrationtests.axt.ActivityScenarioTest.lambda$hasPixelWhenUsingCustomAddingActivityHardwareRendering$16(ActivityScenarioTest.java:419)
	at androidx.test.core.app.ActivityScenario.lambda$onActivity$2$androidx-test-core-app-ActivityScenario(ActivityScenario.java:789)
```

### Robolectric & Android Version

Robolectric master branch 03bbc9c2ac53f2f4a5a334a768ae609f9d06b352 (current latest commit)

**Comment by hoisie** [comment]
Thanks @takahirom for this issue. We will definitely look at this issue. Overall I would say that HW rendering in Robolectric is still experimental. There are currently a lot of issues with it and  the APIs are subject to change.

**Comment by hoisie** [comment]
Hi @takahirom,

@ralf-at-android and I looked at this issue just now. The root cause is that when Activities are created in the manner described in your test, the Window of the Activity does not contain `FLAG_HARDWARE_ACCELERATED`.

If you use this code to set up the Activity, the test passes:
```
    ActivityController<CustomActivity> customActivityController =
        Robolectric.buildActivity(CustomActivity.class);
    customActivityController
        .get()
        .getWindow()
        .setFlags(
            WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED,
            WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED);
    CustomActivity activity = customActivityController.setup().get();
    
```

When adding the Activity to an AndroidManifest.xml, the FLAG_HARDWARE_ACCELERATED flag of ActivityInfo gets properly added I believe.

@ralf-at-android are still considering how to resolve this issue. FLAG_HARDWARE_ACCELERATED is default to true since Android 14, so we this should also be the default for all Activities in Robolectric. However, that may be a breaking change. We will look into this issue more.

**Comment by hoisie** [comment]
This has been fixed and released in 4.12.2.

**Comment by takahirom** [comment]
Thank you for the fix! I edited the comment using `ShadowPackageManager.addActivityIfNotPresent()`
https://github.com/robolectric/robolectric/pull/4736#issuecomment-1831034882
